src/python/pants/backend/python/rules/create_requirements_pex.py

```diff
@@ -54,16 +54,14 @@ def create_requirements_pex(request, pex_bin, python_setup, pex_build_environmen
   argv.extend(interpreter_constraint_args + list(request.requirements))
 
   execute_process_request = MultiPlatformExecuteProcessRequest(
-    (
-      (Platform.current, Platform.current),
-      ExecuteProcessRequest(
+    {
+      (Platform.current, Platform.current): ExecuteProcessRequest(
```

Comment:
The semantics of this are interesting. The effect of using `MultiPlatformExecuteProcessRequest` is to say that the resulting PEX is platform specific (which is true). But I think it's also saying that we can only generate it for our current platform (which is false).

The outputs here _do_ differ per platform, but this command will work on both OSX and Linux.

Reply:
>The outputs here do differ per platform, but this command will work on both OSX and Linux.

We don't allow cross-building yet. When we do we will have to change the platform constraint based on which target platform a user has requests. This does have implications for caching, we will have to add the platform constraints to the protobuf as an extra property (as suggested by @illicitonion earlier). That way the requests won't collide in the cache. We could preemptively do that in this PR or wait for the follow up that introduces the new flags to select platforms. The PR was supposed to be just about not doing the wrong thing, so I think we should wait.

Reply:
I thought about this a bit more, and it occurs to me that you might be concerned about this getting cached on linux in CI and then a remote cache hit causing this to be run or used on an OSX laptop. I think that is a valid concern if we are planning to turn on and demo remote caching before the follow up PR dealing with target platform selection. If so, then we should add the target platform as a property on the request digest so that an osx laptop looking for the PEX output will only get a cache hit for the correct platform.

Reply:
I'm not really thinking about caching.

I'm thinking about the fact that we're claiming that the command itself is only compatible with our current platform, when in reality, the _command_ is cross-platform, but the _output_ of the command is not.

It's fine to leave this as is for now, but I think that we should definitely make sure that we have a story for this planned out, and it would be good to have a TODO here link to something (the design doc shared with pants-devel@ maybe, or a ticket) that describes how we think "remoting the build of a PEX" might work in the future.

Reply:
I read the constraints as "if this command is run on *some platform* the output is only compatible with *some platform*" not a declaration of the commands ability to finish with exit code 0.

Reply:
Ok. But I'm not sure if that will be clear to a reader: especially in this context where we're using `Platform.current` as a bit of a workaround.

At the very least, it needs a longer comment I think.